### PR TITLE
Update patch till 2019-12-31

### PR DESCRIPTION
**Pull request by kei-gbf**
## Update 2019-12-31
- New Characters
  - [x] [Yuisis (Fire)](https://gbf.wiki/Yuisis_(Fire))
        [火ユイシス(gamewith)](https://xn--bck3aza1a2if6kra4ee0hf.gamewith.jp/article/show/180315)
  - [x] [Noa (Grand)](https://gbf.wiki/Noa_(Grand)) リミテッド・ノア
  - [x] [Vikala](https://gbf.wiki/Vikala) ビカラ
  - [x] [Lucius (SSR)](https://gbf.wiki/Lucius_(SSR)) ルシウスSSR
- New Weapons
  - [x] [Gangsta Knife](https://gbf.wiki/Gangsta_Knife) ドス
  - [x] [Ivory Ark](https://gbf.wiki/Ivory_Ark) アーク
  - [x] [Rodentius](https://gbf.wiki/Rodentius) 天干地支筒･子之飾
  - [x] [Alabaster Zenith](https://gbf.wiki/Alabaster_Zenith) 極･白刃
- New Summon
  - [Belial](https://gbf.wiki/Belial) ベリアル
    - nothing new for summon tab,
      Users can use Party's HP buff and supplimental damage up.

## Update 2019-12-28
- New Characters
  - [x] [Helel ben Shalem](https://gbf.wiki/Helel_ben_Shalem) ヘレル・ベン・シャレム
- New Weapons
  - [x] [Bab-el-Mandeb](https://gbf.wiki/Bab-el-Mandeb) バブ・エル・マンデブ 

## This PR does NOT include Events weapons
  - [Red King Crab](https://gbf.wiki/Red_King_Crab) タラヴァガニ
  - [Dawn Rising](https://gbf.wiki/Dawn_Rising) ゴライコウ

本PRに期間途中のイベント武器は含みません。

## Internal changes
- [x] New skill normalCriticalLL (normal critical II) implemented as 12% temporary
- [x] New skill normalSetsunaLL (normal setsuna II)
  (EDIT: ~~rasetsu~~ name mistype only commit log)
- [x] New skill normalOntyouS (but the amount is an empty yet, require verification)
  (EDIT: fix typo ~~Onsyou~~)

通常技巧IIは暫定としてスキルレベルに関わらず効果値 12% として仮実装。
恩寵小はスキル枠の未実装で効果値なし。(要検証)

## Known issues

- [x] Emnity support is stacked? The maximum amount should be applied?

複数の背水サポアビ持ちキャラを編成した時、
効果量が余分に加算されてしまう。(要検証・調整)

### CODE REVIEW

File: scripts/chara_data_converter.py [hunk 1]
```diff
@@ -43,6 +43,9 @@
 sexlist[u"不明"] = "other"
 
 supportAbilist = OrderedDict()
+supportAbilist["none"] = {
+    u"剣聖を継ぐ者", # NOTE: Avoid mis-matches with "剣聖"   
+}
```
Comment (kei-gbf) [inline scripts/chara_data_converter.py]: This is a quick dirty solution. without this "none" entry,
"剣聖を継ぐ者" had matched with "剣聖" (ougi_gage_up_own_20)

ordered-dict ensure the order of processing "none" key at the first iteration.

File: txt_source/charaData.txt [hunk 2]
```diff
@@ -28,6 +28,11 @@
...
+|&attachref(img/3040253000_01.png,25%);|[[ユイシス(火属性ver) (SSR)]]|&color(#ff0000){火};|攻撃|エルーン|女|剣/刀|[[破煌刃・絶影>人物奥義#zetuei]]|&color(#B00000){■};[[鬼哭啾々>人物アビリティ#kikokusyuusyuu]]&br;&color(#B00000){■};[[暁闇>人物アビリティ#gyouan]]&br;&color(#B00000){■};[[一刀命終(条件Lv45)>人物アビリティ#ittoumeisyuu]]|[[墓前の墨染桜>人物サポートアビリティ#sumizomesakura]]&br;[[七回忌の砌>人物サポートアビリティ#nanakaikinomigiri]]|176|1820|''952''|''9640''|[[ドス>武器/ドス (SSR)]]|立花理香|
+|&attachref(img/3040254000_01.png,25%);|[[ルシウス(SSR) (SSR)]]|&color(#800080){闇};|攻撃|ヒューマン|男|剣/刀|[[絶・白刃一掃>人物奥義#zetuhakujin]]|&color(#B00000){■};[[白刃炎斬>人物アビリティ#hakujinenzanssr]]&br;&color(#B00000){■};[[悪鬼無骨>人物アビリティ#akkibukotu]]&br;&color(#B00000){■};[[急・改(条件Lv45)>人物アビリティ#kyuukai]]|[[剣聖を継ぐ者>人物サポートアビリティ#kenseiwotugu]]&br;[[無間斬撃>人物サポートアビリティ#mumazangeki]]|230|1880|''1285''|''9880''|[[極・白刃>武器/極・白刃 (SSR)]]|子安武人|
```
Comment (kei-gbf) [inline txt_source/charaData.txt]: edit "ルシウス(SSR) (SSR)" to be parsed as "ルシウス(SSR)"
The converter script will parse the last rarity

I am not familiar with the naming conversation rule, so please suggest.
(save data will store the name as unique ID, we can't change this later without break backward compatibility)
- not 闇ルシウス, SR dark Lusius already exists. it make us confuse.
- There are SSR + chara-name format

**Comment by kagekiyo7** [comment]
> New skill normalRasetsuLL (normal rasetsu II)

Setsuna?

>New skill normalOnsyouS

Ontyou?
maybe 3%. (同じ効果である庇護と恩寵の効果量が同じだとしたら)
https://twitter.com/konmatsuge/status/1121780651886141441
https://twitter.com/konmatsuge/status/1213012530051727360

**Comment by kei-gbf** [comment]
>> New skill normalRasetsuLL (normal rasetsu II)

> Setsuna?

Yes, my mistake only commit-log and PR comment
TODO: fix commit log on Squash merge

>> New skill normalOnsyouS

> Ontyou?

thx, typo on PR comment, commit log was ok.

> maybe 3%. (同じ効果である庇護と恩寵の効果量が同じだとしたら)
- https://twitter.com/konmatsuge/status/1121780651886141441
- https://twitter.com/konmatsuge/status/1213012530051727360

BTW, 別枠乗算 (Multiplication ?

----
Current head has issue translation key Error 全体背水効果(ザルハメリナ, 火ユイシス)
I will fix it soon

**Comment by kei-gbf** [comment]
- Fixed translation key error
- Add normalOntyouS skill values (unverificated)

File: txt_source/charaData.txt [hunk 3]
```diff
@@ -28,6 +28,11 @@
...
 |&attachref(img/3040169000_01.png,25%);|[[ニーア (SSR)]]|&color(#800080){闇};|特殊|エルーン|女|斧&br;短剣|[[クラーゲン・トーテンタンツ>人物奥義#klagen]]|&color(#cc99ff){■};[[死ト愛ノ世界>人物アビリティ#sitoainosekai]]&br;&color(#CCCC33){■};[[愛シイ人>人物アビリティ#itosiihito]]&br;&color(#CCCC33){■};[[終焉ノ愛>人物アビリティ#syuuennoai]]|[[愛ヲノゾム意志>サポートアビリティ一覧#aiwonozomu]]&br;[[死の逆位置>サポートアビリティ一覧#sinogyakuiti]]&br;[[恋人の正位置>サポートアビリティ一覧#koibitonoseiiti]]|200|1500|''1313''|''8906''|十賢者ページ参照|下屋則子|
+|&attachref(img/3040251000_01.png,25%);|[[ヘレル・ベン・シャレム(リミテッドver) (SSR)]]|&color(#800080){闇};|特殊|不明|女|槍/杖|[[ケイオス・レギオン>人物奥義#chaosregion]]|&color(#B00000){■};[[アバカブ>人物アビリティ#abakav]]&br;&color(#CCCC33){■};[[インヴィジブルタッチ>人物アビリティ#invisibletouch]]&br;&color(#333399){■};[[ロス・エンドス(条件Lv45)>人物アビリティ#losends]]|[[審問と排斥の預言者>人物サポートアビリティ#sinmontohai]]&br;[[ビート・ザ・マウス>人物サポートアビリティ#beatthemouse]]|236|1670|''1270''|''8880''|[[バブ・エル・マンデブ>武器/バブ・エル・マンデブ (SSR)]]|伊瀬茉莉也|
```
Comment (kei-gbf) [inline txt_source/charaData.txt]: master repo had changed 不明 → その他
it's unexpected change, converter script could keep 不明 as unknown
or we have to modify the cell manually every times. wiki still use 不明

**Comment by kagekiyo7** [comment]
> BTW, 別枠乗算 (Multiplication ?

Yes, already done.

**Comment by kagekiyo7** [comment]
> Emnity support is stacked? The maximum amound should be applied?

maybe yes
https://twitter.com/GANBALISM/status/1217246299747012609

**Comment by kei-gbf** [comment]
> maybe yes
> https://twitter.com/GANBALISM/status/1217246299747012609

thx info, but
I think this tries does not proof stacked or not
he did not mention Zahlhamelina only case.

if Zahlhamelina > Yuisis and Zahlhamelina's amount was used then
the result may be same.

**Comment by kei-gbf** [comment]
- HP Full
  - 149197
- Yuisis 20775/28856 (HP 72%)
  - 154625 (≒ 154419 = 149197 x 1.035(HP75%)) ... ok
- Yuisis 20181/28856 (HP 70%)
  Zahlhamelina 20181/30742 (HP 65%)
  - 166677
    - 149197 x 1.1 (HP50%) = 164116 ... 16677 was greater than HP50% case, so stacked?
      - it is within random x1.05-0.095

**Comment by kei-gbf** [comment]
other case info I could found was

https://twitter.com/firemagma/status/730242147283173376
> スキル背水とは乗算、SSRサラのサポアビ背水とは加算。

stacked with SSR Sara (own emnity)

**Comment by Groovman** [comment]
Did the event stuff and support skills that were missing.

**Pull request merged**